modem: hl7800: Update driver to use gpio_dt_spec

galak opened the pull request:
Simplify the implementation by using gpio_dt_spec.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
modem: hl7800: Update driver to use gpio_dt_spec

Simplify the implementation by using gpio_dt_spec.

Signed-off-by: Kumar Gala <galak@kernel.org>

gmarull reviewed (commented):
suggestion

galak commented on drivers/modem/hl7800.c:
```diff
@@ -912,28 +881,21 @@ static void modem_assert_pwr_on(bool assert)
 		HL7800_IO_DBG_LOG("MDM_PWR_ON -> ASSERTED");
-		gpio_pin_set(ictx.gpio_port_dev[MDM_PWR_ON],
-			     pinconfig[MDM_PWR_ON].pin, MDM_PWR_ON_ASSERTED);
+		gpio_pin_set_dt(&hl7800_cfg.gpio[MDM_PWR_ON], MDM_PWR_ON_ASSERTED);
```
@rerickson1 thoughts / comments on this?

rerickson1 replied:
We could remove all the ASSERTED/DE-ASSERTED defines and move everything to device tree as it should be.

galak replied:
Done.

galak committed:
boards: arm: pinnacle_100_dvk: Add GPIO flags for hl7800 to dts

Update devicetree to have GPIO flags set for various pins on the
hl7800 modem.

Signed-off-by: Kumar Gala <galak@kernel.org>

rerickson1 commented on boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts:
```diff
@@ -113,14 +113,14 @@
-		mdm-uart-dsr-gpios = <&gpio0 25 0>;
-		mdm-uart-cts-gpios = <&gpio0 15 0>;
-		mdm-gpio6-gpios = <&gpio1 12 0>;
+		mdm-reset-gpios = <&gpio1 15 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
+		mdm-wake-gpios = <&gpio1 13 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
```
Also, what about `GPIO_ACTIVE_HIGH` or `GPIO_ACTIVE_LOW`

galak replied:
You tell me, the driver itself wasn't setting GPIO_ACTIVE_HIGH or GPIO_ACTIVE_LOW so I just mimic what was in the pinconfig table into the dts.

rerickson1 replied:
should be:
```
mdm-reset-gpios = <&gpio1 15 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
mdm-wake-gpios = <&gpio1 13 (GPIO_OPEN_SOURCE | GPIO_ACTIVE_HIGH)>;
mdm-pwr-on-gpios = <&gpio1 2 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
mdm-fast-shutd-gpios = <&gpio1 14 (GPIO_OPEN_DRAIN | GPIO_ACTIVE_LOW)>;
```

galak replied:
Ok updated the dts and code.  Let me know if that all looks correct or not now.

gmarull commented on boards/arm/pinnacle_100_dvk/pinnacle_100_dvk.dts:
```diff
@@ -113,14 +113,14 @@
-		mdm-uart-cts-gpios = <&gpio0 15 0>;
-		mdm-gpio6-gpios = <&gpio1 12 0>;
+		mdm-reset-gpios = <&gpio1 15 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
+		mdm-wake-gpios = <&gpio1 13 (GPIO_OUTPUT | GPIO_OPEN_DRAIN)>;
```
afaik `GPIO_OUTPUT` is not set in DT, am I wrong?

galak committed:
modem: hl7800: Cleanup gpio assert/deassert

Use logical values of 0/1 for gpio_pin_set_dt and handle any
inversions as GPIO flags that might be required at a board level.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak commented:
CI failures are unrelated to this.  Merging and will look at a PR to fix the CI issue.
